### DISCUSSION TITLE
Generate migration for changes made in Database by other means

### DISCUSSION THREAD

**Discussion by psnehanshu** [prisma/prisma#13447] [Q&A] [answered]
I have a database, which at the moment is at sync with my prisma schema. Suppose I make a change to the database using some other tool (like add new column using ALTER SQL), then I can run `prisma db pull` and it will update the prisma schema to reflect the changes.

Now my question is, can we generate a migration file, which will also record that change? Because, otherwise if I then make changes in the prisma schema, and try to do `prisma migrate dev`, it will prompt a reset.

**Comment by nurul3101** [comment, marked as answer]
Hey @pSnehanshu 👋 
I think you can use the `npx prisma migrate diff` command to create a migration file by comparing your database and the existing prisma schema.
You can learn about the command here: [migrate diff reference](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
